toplevel/vernacentries.ml

```diff
@@ -136,7 +136,10 @@ let make_cases_aux glob_ref =
 		   let n' = Namegen.next_name_away_with_default (Id.to_string Namegen.default_dependent_ident) n avoid in
 		   Id.to_string n' :: rename (n'::avoid) l in
 	     let al' = rename [] al in
-	     (Id.to_string consname :: al') :: l)
+	     let dirpath = Nametab.dirpath_of_global glob_ref in
+	     let consname_qualid = Libnames.make_qualid dirpath consname in
+	     let consname_global = Nametab.locate consname_qualid in
+	     (Libnames.string_of_qualid (Nametab.shortest_qualid_of_global Id.Set.empty consname_global) :: al') :: l)
```

Comment:
One could certainly avoid a new call to `locate`. For instance, building directly a `ConstructRef cstr` global reference from a `IndRef ind` global reference, using `ith_constructor_of_inductive`, rather than by looking at the `consnames`. But this requires to use an index (starting from 1) in the `fold_right`, so this is different approach.

The `Show Match` could also be made more robust, for instance in the presence of `let` in the branch of a constructor. A `decompose_prod_assum` would probably suffice for that, but then, it is a `Context.Rel.Declaration.t` which needs to be matched and not just a pair `(name,type)`.

Otherwise, this is OK. After all, the API on inductive types is not so canonical in general, and in this simple case, working at the low-level of `mind_nf_lc` seems reasonable (compared to working at the higher-level of `Inductiveops`).